app/src/test/kotlin/io/homeassistant/companion/android/widgets/BaseGlanceEntityWidgetReceiverTest.kt

```diff
@@ -242,4 +244,60 @@ class BaseGlanceEntityWidgetReceiverTest {
 
         assertFalse(producer.isActive)
     }
+
+    @Test
+    fun `Given valid intent with ACTION_APPWIDGET_CREATED when onReceive is called then persist WidgetEntity in DAO`() = runTest {
+        val receiver = getReceiver()
+        val context: Context = mockk()
+        val appWidgetId = 42
+        val intent = mockIntent(ACTION_APPWIDGET_CREATED, appWidgetId)
+        val rawEntity = TodoWidgetEntity(-1, 43, "entity")
+        val updatedEntity = TodoWidgetEntity(42, 43, "entity")
+
+        every { intent.getSerializableExtra(EXTRA_WIDGET_ENTITY) } returns rawEntity
+        coJustRun { mockedDao.add(updatedEntity) }
+
+        receiver.onReceive(context, intent)
+
+        advanceUntilIdle()
+
+        coVerify(exactly = 1) { mockedDao.add(updatedEntity) }
+    }
+
+    @Test
+    fun `Given intent with ACTION_APPWIDGET_CREATED without EXTRA_WIDGET_ENTITY when onReceive is called then do nothing`() = runTest {
```

Comment:
Would it make sense to have this test and the next one as a parametrized test instead? Both are "required extras" for the intent and already have almost exactly the same code.

Reply:
It's two different parameter, I prefer to keep them separated to make it clear.